### PR TITLE
Semantic UI components

### PR DESCRIPTION
**Pull request by reshav-abraham**
Replaced material and ag-grid components with semantic ui components

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ An SPA (single-page app) in React that requests information from the API, and:</
 <li>Renders a list or grid of movies</li>
 <li>Allows searching (new API request) and filtering for movies</li>
 <li>Allows clicking on a Movie to see a more detailed view</li>
+<li>https://semantic-ui.com/elements/icon.html</li>
+<li>https://react.semantic-ui.com/usage/</li>
```
Comment (fkotsian) [inline README.md:11]: docs 👍

File: client/components/Grid.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ import 'ag-grid-community/dist/styles/ag-theme-alpine.css';
 /*
 Presentation Component For Grid data
 Pass row columns and data
+Implementated with Ag grid
```
Comment (fkotsian) [inline client/components/Grid.js:9]: 🚿 ?

File: client/components/MovieGrid.js [hunk 3]
```diff
@@ -1,33 +1,62 @@
...
+          }
+    
+      }));
 
    return (
        <div>
-            <Grid rowData={movies} fields={["Movie", "Overview", "Release Date", "Language"]}/>
+            <p></p> 
+            {/* <Grid rowData={movies} fields={["Movie", "Overview", "Release Date", "Language"]}/> */}
+            <SemanticGrid rowData={movies} fields={["Movie", "Overview", "Release Date", "Language"]}/>
```
Comment (fkotsian) [inline client/components/MovieGrid.js:59]: looking good!

File: client/components/SemanticGrid.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+/*
+Grid Presntational component implemented with Semantic UI
+*/
+
+
+
+export const SemanticGrid = props => {
+    return (
+            <Grid style={{height:"700px", width:"1200px", overflow: "scroll"}}>
+                <div class="ui celled grid">
```
Comment (fkotsian) [inline client/components/SemanticGrid.js:13]: ah, `celled grid` 💛

File: client/components/SemanticGrid.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                <Grid.Row>{props.fields.map(field => <Grid.Column width={4}><b>{field}</b></Grid.Column>)}</Grid.Row>
+                {
+                    // ["Movie", "Overview", "Release Date", "Language"]
+                    props.rowData.map(row => 
+                                            <Grid.Row>
```
Comment (fkotsian) [inline client/components/SemanticGrid.js:18]: yep rows and columns

File: client/index.html [hunk 6]
```diff
@@ -3,6 +3,7 @@
 <head>
     <meta charset="UTF-8">
     <title>Movie App</title>
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/semantic-ui/2.4.1/semantic.min.css" />
```
Comment (fkotsian) [inline client/index.html:6]: CDN 👍

File: client/package.json [hunk 7]
```diff
@@ -20,17 +20,27 @@
...
     "@material-ui/core": "^4.11.3",
     "@material-ui/icons": "^4.11.2",
     "@material-ui/pickers": "^3.3.10",
+    "ag-grid-community": "^25.2.1",
     "ag-grid-react": "^25.2.0",
     "axios": "^0.21.1",
     "core-js": "^3.10.0",
     "date-fns": "^2.21.1",
+    "file-loader": "^6.2.0",
+    "node-sass": "^6.0.0",
```
Comment (fkotsian) [inline client/package.json:33]: 👍

File: client/webpack.config.js [hunk 8]
```diff
@@ -10,7 +10,14 @@ module.exports = {
     module : {
         rules : [
             {test : /\.(js)$/, use:'babel-loader'},
-            {test : /\.css$/, use:['style-loader', 'css-loader']}
+            {test : /\.css$/, use:['style-loader', 'css-loader']},
+            {
+                test: /\.(png|jpg|gif|svg|eot|ttf|woff|woff2)$/,
+                loader: 'url-loader',
```
Comment (fkotsian) [inline client/webpack.config.js:16]: whoa, good work. Good ol' icon-loaders

File: client/yarn.lock [hunk 9]
```diff
@@ -3,28 +3,28 @@
 
 
 "@ag-grid-community/react@^25.2.0":
-  "integrity" "sha512-dNESmLUzAGDiKpSznKEVL+HoDULidgkFz24eukNOYRXo/TnsSVJSVfRdPe+be++oLEZLF3WnTjiAuExvIG2fvA=="
-  "resolved" "https://registry.npmjs.org/@ag-grid-community/react/-/react-25.2.0.tgz"
-  "version" "25.2.0"
+  version "25.2.0"
```
Comment (fkotsian) [inline client/yarn.lock:6]: Probably want to delete either this or package-lock and standardize on one of them

**Review by fkotsian — COMMENTED** [review]
Semantic-UI: looking good! How do you like the interface compared to ag-grid?

RE: PR: Can we either:
- separate the Grid work from the useImperativeHandler/forwardRef work;
or
- merge this and unwind the ImperativeHandler/ForwardRef work in the next PR?

left some comments as well!

Looking good so far!

**Comment by reshav-abraham** [comment]
> Semantic-UI: looking good! How do you like the interface compared to ag-grid?
> 
> RE: PR: Can we either:
> 
> * separate the Grid work from the useImperativeHandler/forwardRef work;
>   or
> * merge this and unwind the ImperativeHandler/ForwardRef work in the next PR?
> 
> left some comments as well!
> 
> Looking good so far!

I like Semantic Grid. There's sometimes too much magic with ag-grid. 
I will merge this and undwind the imperative Forward ref stuff!

**Pull request merged**